### PR TITLE
Upgrade cmake files to support build of deb packages.

### PR DESCRIPTION
**Pull request by august-alt**

### CODE REVIEW

File: doc/CMakeLists.txt [hunk 1]
```diff
@@ -2,11 +2,19 @@
 set(ADMC_HELP_PROJECT ${CMAKE_CURRENT_SOURCE_DIR}/admc.qhp)
 set(ADMC_COMPRESSED_HELP ${CMAKE_CURRENT_BINARY_DIR}/admc.qch)
 
-add_custom_command(
-    OUTPUT ${ADMC_COMPRESSED_HELP}
-    COMMAND qhelpgenerator-qt5 ${ADMC_HELP_PROJECT} -o ${ADMC_COMPRESSED_HELP}
-    DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/*
-)
+if (ADMC_BUILD_DEB)
```
Comment (Kvel2D) [inline doc/CMakeLists.txt:5]: Do this instead:

```
find_program(QHELPGENERATOR
    NAMES qhelpgenerator qhelpgenerator-qt5
)
```

And then use `${QHELPGENERATOR}` in the `add_custom_command()` block.

Comment (iv-m) [reply doc/CMakeLists.txt:5]: Please note that in ALT-based systems, `qhelpgenerator` (without `-qt5` suffix) is from Qt4. So, if `qhelpgenerator-qt5` exists, it should be preferred, like this: 

```
find_program(QHELPGENERATOR NAMES qhelpgenerator-qt5 qhelpgenerator)
```

Comment (iv-m) [reply doc/CMakeLists.txt:5]: Btw, for reasonably recent Qt5 versions you can just use `Qt5::qhelpgenerator` target defined in `Qt5Help` module.

Comment (Kvel2D) [reply doc/CMakeLists.txt:5]: Yes, best to use `Qt5::qhelpgenerator` then:

```
find_package(Qt5 REQUIRED
    COMPONENTS
        Help
)
...
add_custom_command(
    OUTPUT ${ADMC_COMPRESSED_HELP}
    COMMAND Qt5::qhelpgenerator ${ADMC_HELP_PROJECT} -o ${ADMC_COMPRESSED_HELP}
    DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/*
)
```

**Pull request merged**